lib/hmmlearn/base.py

```diff
@@ -858,3 +909,434 @@ def _do_mstep(self, stats):
 
 
 _BaseHMM = BaseHMM  # Backcompat name, will be deprecated in the future.
+
+
+class VariationalBaseHMM(BaseHMM):
+
+    def __init__(self, n_components=1,
+                 startprob_prior=None, transmat_prior=None,
+                 algorithm="viterbi", random_state=None,
+                 n_iter=100, tol=1e-6, verbose=False,
+                 params="ste", init_params="ste",
+                 implementation="log"):
+        self.n_components = n_components
+        self.startprob_prior = startprob_prior
+        self.transmat_prior = transmat_prior
+        self.algorithm = algorithm
+        self.random_state = random_state
+        self.n_iter = n_iter
+        self.tol = tol
+        self.verbose = verbose
+        self.params = params
+        self.init_params = init_params
+        self.implementation = implementation
+        # For the case of updating all model components
+        # at each iteration, we can be strict with the convergence
+        # monitory - we know that the Lower Bound will improve
+        # at each iteration
```

Comment:
Why not put the same strict check in the standard BaseHMM?  (Also, are you sure there's no risk of floating point slop causing non-strict improvements, as in BaseHMM?)

Reply:
Variational Algorithms guarantee that the lower bound will increase at each step.  (https://arxiv.org/abs/1601.00670)   During my thesis testing I did not see non-strict improvements due to floating point slop, though I realize that is not a strict test.

I believe in BaseHMM, the way we have priors for model parameters makes the Log likelihood not always strictly decreasing.  If the priors were all equal to zero, then we could add this check.  However, I was not confident about making this assertion there.

Reply:
Sure.  Can you leave a comment to that effect in the source?

Reply:
done.